ui/cypress/e2e/test_model.e2e.cy.js

```diff
@@ -0,0 +1,243 @@
+import 'cypress-iframe';
+import { genBaseModel, generatedIDs } from '../seeds/model_api_data';
+const { faker } = require('@faker-js/faker');
+
+const date = Date.now().toString();
+const directiveCmd = `run file.txt --data=${date}`;
+const modelName = `TestModel_created_at=${date}`;
+
+import isEmpty from 'lodash/isEmpty';
+import get from 'lodash/get';
+import last from 'lodash/last';
+
+import { createModel, shutdownWorker,
+         provisionAndWaitReady,
+         waitForAllUrlsToFinish,
+         getTestModelRegisterUrls,
+         waitUrlToProcess } from '../support/helpers';
+
+/**
+ *
+ **/
+function cleanModel(modelId) {
+
+  cy.visit('/admin');
+
+  return shutdownWorker().then(() => {
+    console.log('Done clearing worker, clearing seeds.');
+    cy.task('seed:clean', {type: 'model', id: modelId});
+  });
+}
+
+
+
+// NOTE slowest test of them all since it registers jataware/test-model
+// and cannot determine how much to wait for the HTML canvas terminal ssh session
+// commands.
+describe('Register/Publish Test Model', { browser: ['chrome', 'chromium', 'firefox'] }, () => {
+
+  let modelId = undefined;
+
+  afterEach(() => {
+    cleanModel(modelId);
+  });
+
+  it('E2E jataware/test-model Register/Publish completed.', () => {
+
+    // in case we want to reuse/override a test with forced ID:
+    // modelId = 'acab2a55-8356-4029-99c2-734b4939293e';
+
+    const modelOverrides = {
+      is_published: false,
+      commit_message: '',
+      period: null,
+      outputs: [],
+      image: '',
+      qualifier_outputs: [],
+      geography: null
+    };
+
+    const testModel = createModel(modelId, modelOverrides);
+
+    modelId = testModel.id;
+
+    console.log('Test Model ID:', modelId);
+
+    cy.wrap(provisionAndWaitReady(modelId))
+      .then(() => {
+
+        const directiveParamValue = 4.8;
+        const fileName = `output_0.9_${directiveParamValue}.csv`;
+        const folderName = 'test-model';
+        const user = 'clouseau';
+        const homeDir = `/home/${user}`;
+
+        const saveUrl = `/api/terminal/container/${testModel.id}/ops/save?path=/home/clouseau/${folderName}/${fileName}`;
+
+        cy.visit(`/term/${testModel.id}`);
+
+        cy.reload();
+
+        cy.findByRole('textbox', {name: /terminal input/i}).as('TerminalInput');
+
+        cy.wait(500);
+
+        cy.get('@TerminalInput')
+          .type(`sudo apt update && sudo apt install -y python3-pip && git clone https://github.com/jataware/test-model.git && cd test-model{enter}`, {force: true});
+
+        cy.wait(25000); // 60000 for slower computers
```

Comment:
This has a bunch of overlap with `model.e2e.cy.js` (not missing `test_` prefix on file). This particular file tests registering our jataware/test-model, which means we have to type on the virtual xterm terminal and wait for long commands to finish (sudo apt update; sudo apt install ...; pip install ...). I don't know of a proper way to verify once a command in this ssh-virtual terminal is completed, so we have arbitrary waits here. You'll have to change it in order to work on your system. This test is a bit overkill, but really auto-tests registering the test model. I still recommend running all the model.e2e.cy.js tests (and component tests!) as it really tests all aspects of model registration piecemeal.

Reply:
Help needed in order to check when the last running command has completed, without waits?

Maybe:

```
cy.get('@TerminalInput')
// select ^ this element, but with a not:disabled check, and add a very long timeout for it? Not great, but maybe better.
```